core/arch/riscv64/emit_utils.c

```diff
@@ -115,8 +129,41 @@ void
 patch_branch(dr_isa_mode_t isa_mode, cache_pc branch_pc, cache_pc target_pc,
              bool hot_patch)
 {
-    /* FIXME i#3544: Not implemented */
-    ASSERT_NOT_IMPLEMENTED(false);
+    /* Compute offset as unsigned, modulo arithmetic. */
+    ptr_int_t off = (ptr_uint_t)target_pc - (ptr_uint_t)branch_pc;
```

Comment:
Perhaps add an assert to make sure that the off is smaller than what can be handled by the branch instruction.

Reply:
It's asserted at line 142 and line 153. J-type and B-type have different reachability, so it's separated.